build-scan-shared/__tests__/summary/dump.test.ts

```diff
@@ -0,0 +1,141 @@
+import {Job} from '../../src/metadata/load'
+
+process.env['RUNNER_TEMP'] = '/tmp'
+
+import * as githubUtils from '../../src/utils/github'
+import * as input from '../../src/setup/input'
+import * as io from '../../src/utils/io'
+import * as load from '../../src/metadata/load'
+import * as output from '../../src/summary/dump'
+import {BuildToolType} from '../../src/buildTool/common'
+
+const outputMock = jest.spyOn(output, 'dump')
+
+const job = {
+    prNumber: 42,
+    builds: [
+        {
+            projectId: 'projectId',
+            workflowName: 'workflow',
+            jobName: 'job',
+            buildToolVersion: '42',
+            requestedTasks: 'install',
+            buildId: 'abcde',
+            buildFailure: false,
+            buildTimestamp: '0',
+            isRepublished: false
+        }
+    ],
+    buildToolType: BuildToolType.MAVEN
+}
+
+describe('dump', () => {
+    afterEach(() => {
+        jest.clearAllMocks()
+    })
+
+    it('Dump triggers pull request comment and summary', async () => {
```

Comment:
what is `dump` in this context? is it the build scan dump?
also is there a test for the conditional build scan publication?

Reply:
`dump` is the name of the function under test

Reply:
> also is there a test for the conditional build scan publication?

by conditional build scan publication, I'm assuming you are referring to `ON_DEMAND` / `ON_FAILURE` flags. Those flags are just used to condition unpublished build scans capture.

Reply:
I have a higher level test in the extension code

Reply:
got it.